### PR TITLE
First pass at traefik/dagster config

### PR DESCRIPTION
**Pull request by shaidar**
# What are the relevant tickets?

https://github.com/mitodl/ol-infrastructure/issues/1473
# Description (What does it do?)

Removes the existing nginx config and replaces it with traefik and traefik-forward-auth in order to leverage our keycloak work and start using OIDC to authenticate to Dagster. More details have been added to the issue linked above as to some of the pain points encountered/tested.
# Screenshots (if appropriate):

- [ ] Desktop screenshots
- [ ] Mobile width screenshots

# How can this be tested?

This has been tested manually by me setting up a dagster, traefik, and keycloak instances and trying out the authentication process.
# Additional Context

- [ ] Add client app in Keycloak (pulumi code)
- [ ] Add secret values in Vault in before merging

### CODE REVIEW

File: src/bilder/images/dagster/templates/docker-compose.yaml.j2 [hunk 1]
```diff
@@ -46,13 +48,38 @@ services:
...
+      - "443:443"
+      - "8080:8080"
     volumes:
-    - {{ context.certificate_file }}:/etc/nginx/nginx.pem:ro
-    - {{ context.certificate_key_file }}:/etc/nginx/nginx-key.pem:ro
-    - {{ context.nginx_directory }}/nginx.conf:/etc/nginx/conf.d/dagster.conf:ro
-    - {{ context.nginx_directory }}/htpasswd:/etc/nginx/htpasswd:ro
+      - "{{ context.traefik_directory}}/traefik.yaml:/etc/traefik/traefik.yaml:ro"
+      - {{ context.certificate_file }}:/etc/traefik/odl-wildcard.pem:ro
+      - {{ context.certificate_key_file }}:/etc/traefik/odl-wildcard-key.pem:ro
```
Comment (Ardiea) [inline src/bilder/images/dagster/templates/docker-compose.yaml.j2:73]: Maybe wrap these two lines in double quotes just so they are the same as 71 and 74?

**Review by Ardiea — APPROVED** [review]

File: src/bilder/images/dagster/files/.env.tmpl
Comment (blarghmatey) [inline src/bilder/images/dagster/files/.env.tmpl:1]: These changes actually need to be preserved. This is for Dagster to be able to use basic auth against the Airbyte service.

File: src/bilder/images/dagster/templates/docker-compose.yaml.j2 [hunk 2]
```diff
@@ -46,13 +48,38 @@ services:
     - "{{ context.edx_pipeline_definition_directory }}:/etc/dagster/:ro"
     env_file:
       - .env
-  nginx:
-    image: nginx
+  traefik:
+    image: traefik:latest
```
Comment (blarghmatey) [inline src/bilder/images/dagster/templates/docker-compose.yaml.j2:52]: Should we use latest, or pin to the current major release version?

File: src/bilder/images/dagster/files/.env_traefik_forward_auth.tmpl
Comment (blarghmatey) [inline src/bilder/images/dagster/files/.env_traefik_forward_auth.tmpl:1]: It looks like this still needs to be uploaded and added to consul-template rendering in the deploy.py logic.

**Review by blarghmatey — APPROVED** [review]

**Pull request merged**